lib/miq_automation_engine/engine/miq_ae_builtin_method.rb

```diff
@@ -160,5 +171,49 @@ def self.category_for_key(obj, key)
       end
     end
     private_class_method :category_for_key
+
+    def self.prepend_vendor(obj)
+      vendor = nil
+      ATTRIBUTE_LIST.detect { |attr| vendor = detect_vendor(obj.workspace.root[attr], attr) }
+      if vendor
+        $miq_ae_logger.info("Setting prepend_namespace to: #{vendor}")
+        obj.workspace.prepend_namespace = vendor
+      end
+    end
+    private_class_method :prepend_vendor
+
+    def self.detect_vendor(src_obj, attr)
+      case attr
+      when "orchestration_stack"
+        src_obj.type.split('::')[2] if src_obj
+      when "miq_host_provision"
+        "vmware" if src_obj
+      when "miq_request", "miq_provision", "vm_migrate_task"
+        src_obj.source.vendor if src_obj && src_obj.source.respond_to?(:vendor)
+      when "vm"
+        src_obj.vendor if src_obj && src_obj.respond_to?(:vendor)
+      end
+    end
+
+    def self.emsevent?(event_stream)
+      event_stream.object_class.name.casecmp("emsevent").zero?
+    end
+    private_class_method :emsevent?
+
+    def self.miq_event_router(obj, _attr)
+      root = obj.workspace.root
+      event_stream = root['event_stream']
+      if event_stream.nil?
+        root['event_path'] = "/System/Event"
```

Comment:
@tinaafitz Should we raise an ArgumentError if we don't get the event_stream object